File: service_tutoral/src/hajime_client.cpp
#include <rclcpp/rclcpp.hpp>
#include <new_message/srv/ningen.hpp>
#include <chrono>
#include <cstdlib>
#include <memory>
 
using namespace std::chrono_literals;

int main(int argc, char** argv)
{
    rclcpp::init(argc, argv);
    if (argc != 3) {
        RCLCPP_INFO_STREAM(rclcpp::get_logger("rclcpp"), "usage: add_two_ints_client X Y");
    }

    std::shared_ptr<rclcpp::Node> node = rclcpp::Node::make_shared("saikoudesu");
    rclcpp::Client<new_message::srv::Ningen>::SharedPtr client = 
            node->create_client<new_message::srv::Ningen>("new_service");
    auto request = std::make_shared<new_message::srv::Ningen::Request>();
    request->a = atoi(argv[1]);
    request->b = atoi(argv[2]);
    while (!client->wait_for_service(1s)) {
        if (!rclcpp::ok()) {
            RCLCPP_ERROR(rclcpp::get_logger("rclcpp"), "interrupted while waiting");
            return 0;
        }
        RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "service not available");
    }
    
    auto result = client->async_send_request(request);
    if (rclcpp::spin_until_future_complete(node, result) ==
            rclcpp::FutureReturnCode::SUCCESS)
    {
        RCLCPP_INFO(rclcpp::get_logger("rclcpp"), result.get()->answer);
        auto param_client = std::make_shared<rclcpp::SyncParametersClient>(node, "hayaku");
        auto as = param_client->get_parameters({"tsuchida_parametor"});
        RCLCPP_INFO(rclcpp::get_logger("rclcpp"), as[0].value_to_string());
        
    }
    else {
        RCLCPP_ERROR(rclcpp::get_logger("rclcpp"), "Fail to service");
    }
    rclcpp::shutdown();
    return 0;
}

File: service_tutoral/scripts/hajime_client.py
import sys 
from new_message.srv import Ningen
import rclpy
from rclpy.node import Node

class MinimalClient(Node):
    def __init__(self):
        super().__init__("minimal_client")
        self.client1 = self.create_client(Ningen, 'tsuchida_service')
        while not self.client1.wait_for_service(timeout_sec=1.0):
            self.get_logger().info('service not available, waiting again...')
        self.req = Ningen.Request()
    
    def send_request(self):
        self.req.a = int(sys.argv[1])
        self.req.b = int(sys.argv[2])
        self.future = self.client1.call_async(self.req)
    
def main(args=None):
    rclpy.init(args=args)
    minimal_client = MinimalClient()
    minimal_client.send_request()
    print("f1")
    while rclpy.ok():
        print("f2")
        rclpy.spin_once(minimal_client)
        print("f3")
        if minimal_client.future.done():
            try:
                response = minimal_client.future.result()
            except Exception as e:
                minimal_client.get_logger().info('Service call failed %r' % (e, ))
            else:
                minimal_client.get_logger().info(
                    'Result of servoce: for %d + %d = %s' %
                    (minimal_client.req.a, minimal_client.req.b, response.answer))
                break
    minimal_client.destroy_node()
    rclpy.shutdown()

if __name__=='__main__':
    main()
            